### ISSUE TITLE
Koin logger is global

### ISSUE THREAD

**Issue by IRus** [InsertKoinIO/koin#421]
**Is your feature request related to a problem? Please describe.**
In case if I create two Koin applications in single JVM instance, I can't have different loggers for different applications. This true for both 1.x, and 2.x Koin.

**Describe the solution you'd like**
Create `logger` property in `KoinApplication`, and point it by default to global logger `KoinApplication.Companion.logger`. Add API to change instance logger.

**Describe alternatives you've considered**
N/A

**Target Koin project**
koin-core

**Comment by IRus** [comment]
Example with koin 1.0:

```
class Holder(var value: Int)

fun main(args: Array<String>) {
    val module = module {
        single<Holder> { Holder(0) }
    }

    Koin.logger = Slf4jKoinLogger

    val koin1 = Koin.create().loadModules(listOf(
        module
    ))

    val koin2 = Koin.create().loadModules(listOf(
        module
    ))

    koin2.logger = MyCustomLogger() // compile error

    koin1.koinContext.get<Holder>().value = 1
    koin2.koinContext.get<Holder>().value = 2

    println(koin1.koinContext.get<Holder>().value)
    println(koin2.koinContext.get<Holder>().value)
}
```

**Comment by arnaudgiuliani** [comment]
sure, the logger is static for now.  It has been an easy way to provide it to all internals, I admit.  If you have a good idea don`t hesitate ;)

**Comment by IRus** [comment]
I see, I've written substitute logger, which calls delegate to log, and tried to pass it around. 
Got problems with `Module`, since it doesn't tie to `Koin` or `KoinApplication`, there are big API changes required to make it proper way.

**Comment by arnaudgiuliani** [comment]
saw that also :/

**Comment by IRus** [comment]
So I see two possible resolutions here: 
1. Make this breaking change part of 2.0 release, since it already have some breaking changes;
2. Close this issue, since it's not an issue for most use cases.

**Comment by arnaudgiuliani** [comment]
not for 2.0.0 release, but we can talk about a futur solution

**Comment by arnaudgiuliani** [comment]
@IRus you can check on the master branch if you want to propose something

**Comment by arnaudgiuliani** [comment]
This has been fixed in current 2.1.0-alpha-8
